pkgs/development/python-modules/mpris_server/default.nix

```
@@ -0,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, emoji
+, pydbus
+, pygobject3
+, unidecode
+, setuptools
+,
+}:
+buildPythonPackage rec {
+  pname = "mpris_server";
+  version = "0.4.2";
+  format = "pyproject";
+
+  src = fetchPypi {
+    pname = "mpris_server";
+    inherit version;
+    hash = "sha256-p3nM80fOMtRmeKvOXuX40Fu9xH8gPgYyneXbUS678fE=";
+  };
```

```suggestion
  src = fetchFromGitHub {
    owner = "alexdelorenzo";
    repo = pname;
    rev = "v${version}";
    hash = "sha256-9VEok2eiGKOQDUxBC29z7Y/bQwZc5Ts1prKJMEY7pGs=";
  };
```

version 0.4.3 is not on PyPi but is the latest on github. the only difference to 0.4.2 is a 1 line change that fixes a typo (MUTE_VAL does not exits) that can result in a NameError exception.

```
diff -r mpris_server-0.4.2/mpris_server/interfaces/player.py mpris_server-0.4.3/mpris_server/interfaces/player.py
365c365
<     elif value == MUTE_VAL:
---
>     elif value == MUTE_VOL:
```

---

I think it would be okay to wait until this is uploaded to PyPI.

---

it doesn't seem like updates are being made on PyPi anymore. this was released july 8 2021.

---

Oh, odd. If that's right, I'll have it use the github.

---

it looks like they pulled 0.4.3 for something unrelated to a failure in another package. probably just fine to leave as is. github is missing tags for all versions other than 0.4.3, so who knows if the 0.4.3 tag will get deleted at some point too.

---

So does that mean I shouldn't use github?

---

up to you.

---

Ok ill leave it with PyPI then